### PR TITLE
snm3C: Optimize first four tasks (including demultiplex)

### PR DESCRIPTION
**Pull request by aawdeh**
- Merged + optimized the first three tasks (after demultiplex) : Merged for loops + running multiple multithreaded instances + tar instead of pigz
- Add multithreading to cutadapt in demultiplex and switched tar to pigz 
- Added error messages to ensure no cells are being dropped

- Changed "snM3C" to "snm3C"
- Updated docker to reflect changes from Wubin (new docker was built to tackle error Wubin mentioned)

### Checklist 
If you can answer "yes" to the following items, please add a checkmark next to the appropriate checklist item(s) **and** notify our WARP documentation team by tagging either @ekiernan or @kayleemathews in a comment on this PR.

- [x] Did you add inputs, outputs, or tasks to a workflow?
- [x] Did you modify, delete or move: file paths, file names, input names, output names, or task names?
- [x] If you made a changelog update, did you update the pipeline version number?

### CODE REVIEW

**Review by ekiernan — APPROVED** [review]
Thank you for all the work and for walking through it!

**Review by kevinpalis — APPROVED** [review]

**Comment by aawdeh** [comment]
Removed scientific test json for miseq data as scientific test was failing on miseq. 

Further investigation of error is required. Tickets have been made to tackle this issue. (https://broadworkbench.atlassian.net/browse/PD-2570, https://broadworkbench.atlassian.net/browse/PD-2506 and https://broadworkbench.atlassian.net/browse/PD-2505) 

Scientific tests for miseq data will be added back once we address the issue.

File: website/docs/Pipelines/snM3C/README.md [hunk 1]
```diff
@@ -159,19 +163,21 @@ The following table lists the output variables and files produced by the pipelin
...
 
-Please cite the following publication the snM3C pipeline: 
+Please cite the following publications the snm3C pipeline: 
 
 Lee, DS., Luo, C., Zhou, J. et al. Simultaneous profiling of 3D genome structure and DNA methylation in single human cells. Nat Methods 16, 999–1006 (2019). https://doi.org/10.1038/s41592-019-0547-z
 
-Please also consider citing our preprint:
+Liu, H., Zhou, J., Tian, W. et al. DNA methylation atlas of the mouse brain at single-cell resolution. Nature 598, 120–128 (2021). https://doi.org/10.1038/s41586-020-03182-8
+
+Also cite our preprint:
```
Comment (kayleemathews) [inline website/docs/Pipelines/snM3C/README.md:180]:
```suggestion
Please cite the following preprint for the WARP repository and website:
```

File: website/docs/Pipelines/snM3C/README.md [hunk 2]
```diff
@@ -159,19 +163,21 @@ The following table lists the output variables and files produced by the pipelin
...
+## Citing the snm3C Pipeline
 
-If you use the snM3C Pipeline in your research, please identify the pipeline in your methods section using the [snM3C SciCrunch resource identifier](https://scicrunch.org/resources/data/record/nlx_144509-1/SCR_025041/resolver?q=SCR_025041&l=SCR_025041&i=rrid:scr_025041).
+If you use the snm3C Pipeline in your research, please identify the pipeline in your methods section using the [snm3C SciCrunch resource identifier](https://scicrunch.org/resources/data/record/nlx_144509-1/SCR_025041/resolver?q=SCR_025041&l=SCR_025041&i=rrid:scr_025041).
 
-* Ex: *snM3C Pipeline (RRID:SCR_025041)*
+* Ex: *snm3C Pipeline (RRID:SCR_025041)*
 
-Please cite the following publication the snM3C pipeline: 
+Please cite the following publications the snm3C pipeline:
```
Comment (kayleemathews) [inline website/docs/Pipelines/snM3C/README.md:174]:
```suggestion
Please cite the following publications for the snm3C pipeline: 
```

**Comment by aawdeh** [comment]
retest this please

**Comment by ekiernan** [comment]
Plumbing test only faied for snm3C here: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/71050/console

We compared the output mapping summaries manually and the only differences are in new columns with cemba_data v1.8.0:

R1SplitReadsUniqueMappedReads is NOT IN truth Columns
R1SplitReadsUniqueMappingRate is NOT IN truth Columns
R1SplitReadsMultiMappedReads is NOT IN truth Columns
R1SplitReadsMultiMappingRate is NOT IN truth Columns
R1SplitReadsOverallMappingRate is NOT IN truth Columns
R2SplitReadsUniqueMappedReads is NOT IN truth Columns
R2SplitReadsUniqueMappingRate is NOT IN truth Columns
R2SplitReadsMultiMappedReads is NOT IN truth Columns
R2SplitReadsMultiMappingRate is NOT IN truth Columns
R2SplitReadsOverallMappingRate is NOT IN truth Columns

**Comment by ekiernan** [comment]
retest this please

**Comment by ekiernan** [comment]
scientific test: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/71279/console

**Pull request merged**